src/config.c

```diff
@@ -1813,7 +1813,7 @@ static int boolConfigSet(standardConfig *config, sds *argv, int argc, const char
         *(config->data.yesno.config) = yn;
         return 1;
     }
-    return 2;
+    return (config->flags & VOLATILE_CONFIG) ? 1 : 2;
```

Comment:
maybe the right thing is to modify the `if (prev != yn) {` line?
so that we still call the module callback?

although, arguably, it's better if the module will implement the apply callback rather than rely on the setter callback..

Reply:
This is a good question. Here's what I think:
Since modules can only have typed configs (enum/bool/number/string) and there's no facility for generic undefined configs (there's nothing equivalent to `createSpecialConfig`) then it's okay to assume that any "apply" functionality should happen in the "apply" phase. In our special configs we have examples of configurations we don't store anywhere and apply-on-set with no apply function. This is ok because they are never interdependent. Such configs, are implicitly volatile and don't need the flag (ex. `dir` config). Modules don't have such cases, so we can assume that if there's anything to apply we need to supply a explicit apply function.

Reply:
arguably, the string type is the module's "special" config type.
i.e. a module that would want to do any special processing (like string splitting) would use that.
but i don't see why that (the special type) is related to this discussion anyway, since the volatile configs we have are plain strings, not special ones.

what we could argue though is that we add this flag mainly for backwards compatibility, so for modules there's no such concern. and also that it doesn't really matter where we place that `if`, as long as we don't expose the VOLATILE flag to the module api, and let them create such configs.

so what we really should be asking ourselves, is if we want to expose the VOLATILE flag to modules.

Reply:
The reason I mentioned the special config is that non-special configs use generic setter logic, and the generic setter logic compares the old value (string/bool/enum...) to the new one and skips the setter. So anyone using the generic setters can take this into account.

I think that since this is was prompted because of backward compatibility we can avoid exposing VOLATILE for now and change it in the future if needed.

Reply:
ok, but in some way all module configs are "special" configs, since they have a setter callback which we don't know what it does.

anyway, i think we can agree not to expose this to modules for now, and when we do, we'll want to move that `if`.
unless we fear that when they day come we'll forget to move it, and prefer to do that now..
i don't mind either way.